core/ledger/ledgermgmt/ledger_mgmt.go

```diff
@@ -132,6 +167,26 @@ func (m *LedgerMgr) CreateLedgerFromSnapshot(snapshotDir string) (ledger.PeerLed
 	}, cid, nil
 }
 
+func (m *LedgerMgr) setUnderCreationLedger(snapshotDir string) error {
+	m.underCreationLock.Lock()
+	defer m.underCreationLock.Unlock()
+	if m.underCreationLedger != "" {
+		return errors.New("a ledger is being created from a snapshot. Call 'peer channel joinbysnapshotstatus' to check the status and execute the command again after it is done.")
+	}
+	ledgerID, err := kvledger.GetChannelNameFromSnapshot(snapshotDir)
+	if err != nil {
+		return err
+	}
+	m.underCreationLedger = ledgerID
```

Comment:
I guess that it's better to add a check and return error if `ledgerID == ""`. On the same note, as far as I recall, there were other restrictions on the channelName... good to verify on the safer side.

Reply:
Since this PR has been updated to not extract ledgerID, the channelID validation will be handled separately.